src/libraries/System.IO.Compression/src/System/IO/Compression/DeflateZLib/DeflateStream.cs

```diff
@@ -233,12 +314,38 @@ public override int ReadByte()
+        /// [!code-vb[System.IO.Compression.DeflateStream#3](~/samples/snippets/visualbasic/VS_Snippets_CLR_System/system.io.compression.deflatestream/vb/program3.vb#3)]
+        /// ]]></format></example>
+        /// <exception cref="System.ArgumentNullException"><paramref name="buffer" /> is <see langword="null" />.</exception>
+        /// <exception cref="System.InvalidOperationException">The <see cref="System.IO.Compression.CompressionMode" /> value was <see langword="Compress" /> when the object was created.
+        /// -or-
+        /// The underlying stream does not support reading.</exception>
+        /// <exception cref="System.ArgumentOutOfRangeException"><paramref name="offset" /> or <paramref name="count" /> is less than zero.
+        /// -or-
+        /// <paramref name="buffer" /> length minus the index starting point is less than <paramref name="count" />.</exception>
+        /// <exception cref="System.IO.InvalidDataException">The data is in an invalid format.</exception>
+        /// <exception cref="System.ObjectDisposedException">The stream is closed.</exception>
         public override int Read(byte[] buffer, int offset, int count)
         {
             ValidateBufferArguments(buffer, offset, count);
             return ReadCore(new Span<byte>(buffer, offset, count));
         }
 
+        /// <summary>Reads a sequence of bytes from the current Deflate stream into a byte span and advances the position within the Deflate stream by the number of bytes read.</summary>
+        /// <param name="buffer">A region of memory. When this method returns, the contents of this region are replaced by the bytes read from the current source.</param>
+        /// <returns>The total number of bytes read into the buffer. This can be less than the number of bytes allocated in the buffer if that many bytes are not currently available, or zero (0) if the end of the stream has been reached.</returns>
+        /// <remarks>Use the <see cref="System.IO.Compression.DeflateStream.CanRead" /> property to determine whether the current instance supports reading. Use the <see cref="O:System.IO.Compression.DeflateStream.ReadAsync" /> method to read asynchronously from the current stream.
+        /// This method read a maximum of `buffer.Length` bytes from the current stream and store them in <paramref name="buffer" />. The current position within the Deflate stream is advanced by the number of bytes read; however, if an exception occurs, the current position within the Deflate stream remains unchanged. This method will block until at least one byte of data can be read, in the event that no data is available. `Read` returns 0 only when there is no more data in the stream and no more is expected (such as a closed socket or end of file). The method is free to return fewer bytes than requested even if the end of the stream has not been reached.
```

Comment:
Need to double check the cref on VS for the Span argument:

```suggestion
        /// This method read a maximum of <c>buffer.Length</c> bytes from the current stream and store them in <paramref name="buffer" />. The current position within the Deflate stream is advanced by the number of bytes read; however, if an exception occurs, the current position within the Deflate stream remains unchanged. This method will block until at least one byte of data can be read, in the event that no data is available. <see cref="System.IO.Compression.DeflateStream.Read(System.Span<System.Byte>)" /> returns 0 only when there is no more data in the stream and no more is expected (such as a closed socket or end of file). The method is free to return fewer bytes than requested even if the end of the stream has not been reached.
```